src/utilities/xml/XMLValidator.hpp

```diff
@@ -0,0 +1,125 @@
+*  OpenStudio(R), Copyright (c) 2008-2022, Alliance for Sustainable Energy, LLC, and other contributors. All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+*  following conditions are met:
+*
+*  (1) Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+*  disclaimer.
+*
+*  (2) Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+*  disclaimer in the documentation and/or other materials provided with the distribution.
+*
+*  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef UTILITIES_XML_XMLVALIDATOR_HPP
+#define UTILITIES_XML_XMLVALIDATOR_HPP
+
+#include "../UtilitiesAPI.hpp"
+
+#include "../core/Path.hpp"
+#include "../core/Logger.hpp"
+#include "../core/LogMessage.hpp"
+#include "../core/Optional.hpp"
+#include "../core/StringStreamLogSink.hpp"
+
+#include <boost/optional.hpp>
+
+#include <string>
+
+namespace openstudio {
+
+enum class XMLValidatorType
+{
+  XSD,
+  XSLTSchematron,
+  Schematron
+};
+
+namespace detail {
+  struct XMLInitializer;
+}
+
+class UTILITIES_API XMLValidator
+{
+ public:
+  /** @name Constructors */
+  //@{
+
+  /// Constructor for a new validator
+  explicit XMLValidator(const openstudio::path& schemaPath);
```

Comment:
I changed this to load the schema only, not the xmlPath. So you can load the GBXML XSD for eg, and then call validate on a bunch of gbXML files if you want, without having to do all that logic again and again.

Reply:
Hmm, I thought we intentionally passed schemaPath AND xmlPath so that you'd know which xml file potential errors pertain to for a single instance of XMLValidator...

Reply:
I don't think that provides much value. You can always check the xmlPath() getter, and if you're going to be checking errors() and warnings() you'd want to do that after the validate call.

eg

```ruby
xmlValidator = XMLValidator.new(schemaPath)
gbxmls.each do |gbxml|
    if !xmlValidator.validate(gbxml)
       puts "#{gbxml} is not valid!"
       puts xmlValidator.errors.map{|l| l.logMessage}
    endif
endif
```

Also, this is exactly the same we do with all the RT/FT classes we have such as EnergyPlusForwardTranslator.

Reply:
(if you really wanted to not let the user validate several xmls with the same schema, then the Ctor should probably call validate() itself (and make it private...), and you'd only provide getters after that.)

Reply:
So you're saying that calling `validate(gbxml)` for the second time clears out any errors thrown from the first call?

Reply:
yes. See the call to reset() atop the validate()

https://github.com/NREL/OpenStudio/blob/558928419c218aa0d59d136bddae24a304abfb8e/src/utilities/xml/XMLValidator.cpp#L221-L234